test/export/test_export.py

```diff
@@ -3744,17 +3744,91 @@ def forward(self, x):
         for op in ops:
             self.assertIn(op, (torch.ops.aten._to_copy.default,))
 
-    def test_device_to_mutation(self):
+    def test_device_to_mutation_return(self):
         class Module(torch.nn.Module):
             def forward(self, x):
                 y = x.to("cpu")
                 y.add_(1)
                 return y, x
 
+        torch.export.export_for_training(Module(), (torch.tensor(1, device="cpu"),))
+
         with self.assertRaisesRegex(
-            RuntimeError, "cannot mutate tensors with frozen storage"
+            RuntimeError, "Cannot return on mutated frozen tensor"
         ):
-            export(Module(), (torch.tensor(1, device="cpu"),))
+            torch.export.export_for_inference(
+                Module(), (torch.tensor(1, device="cpu"),)
+            )
+        # with self.assertRaisesRegex(
+        #     RuntimeError, "Cannot return on mutated frozen tensor"
+        # ):
+        #     export(
+        #         Module(), (torch.tensor(1, device="cpu"),)
+        #     )
+
+    def test_device_to_mutation_return_float(self):
```

Comment:
Can we have a one test that forces copy? I think you can initialize int tensor and call float() on it.

Reply:
> Can we have a one test that forces copy? I think you can initialize int tensor and call float() on it.

i think the `test_device_to_mutation_not_aliasing` test should forces copy? We call `z.to(torch.int64)` on a float.

Reply:
Ah gotcha